### PR TITLE
Code Review 2 Suggestions

### PR DESCRIPTION
**Pull request by JellyV**
**Analysis of the program.**
This program helps you find nearby recycling locations for various items.

**Was the program available in UC Github on time?**
Yes

**Is the program documented/commented well enough for you to understand?**
Yes

**Does the program compile?**
No, but it was just a few curly brackets needed to be fixed.

**Rationale behind my changes.**
I fixed the curly brackets to get the program to build, and after that fixed some code convention violations, typos, and other minor mistakes. The general flow of the program is good, and I didn't make any major changes to the logic/file structure.

**My Commits in my project (I linked 4 as these are interrelated):**
[UI Changes](https://github.com/JellyV/FreeWays/commit/59d5d0a21de04878a4948761245f43c00a029ae2)
[Add Email Authentication](https://github.com/JellyV/FreeWays/commit/471370e07ce85416ad51d0c50d17de853748ae75)
[Added protected user directories](https://github.com/JellyV/FreeWays/commit/e84d5cc8ff60251ac8f5e160aa79f7c4374ebb23)
[Added Google Authentication](https://github.com/JellyV/FreeWays/commit/31014954f80adba85c2d0849679191ceec4d1c66)

**Things I learned**
- I understood Kotlin scoped functions, such as let and it better.
- I learned how to implement authentication using firebase easily, without wasting effort on creating a custom solution.
- I learned more Kotlin best conventions and understood when to avoid things like explicit arguments, val vs var, and so on.

### CODE REVIEW

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 1]
```diff
@@ -29,84 +30,94 @@ class MainActivity : ComponentActivity() {
...
-}
-
-
-private fun signInResult(result: FirebaseAuthUIAuthenticationResult) {
-    val response = result.idpResponse
-    if (result.resultCode == ComponentActivity.RESULT_OK) {
-        var user = FirebaseAuth.getInstance().currentUser
-    }
\ No newline at end of file
+}
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:123]: Git thinks I changed the whole file but I only removed/replaced some brackets so the program would build without errors.

File: app/src/main/java/com/findmyrecycling/MainViewModel.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ class MainViewModel(var productService : ProductService = ProductService()) : Vi
 
     fun fetchProducts() {
         viewModelScope.launch {
-            var innerProducts = productService.fetchProducts()
+            val innerProducts = productService.fetchProducts()
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/MainViewModel.kt:17]: innerProducts doesn't need to be mutable so changed to val.

File: app/src/main/java/com/findmyrecycling/ProfileScreen.kt [hunk 3]
```diff
@@ -5,7 +5,6 @@ import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.compose.foundation.border
 import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Spacer
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/ProfileScreen.kt:8]: removed unused import

File: app/src/main/java/com/findmyrecycling/dto/Product.kt [hunk 4]
```diff
@@ -3,8 +3,9 @@ package com.findmyrecycling.dto
 import com.google.gson.annotations.SerializedName
 
 data class Product(
-    var product : String,
-    var id : Int = 0,) {
+    internal var product : String,
+    internal var id : Int = 0,
+) {
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/dto/Product.kt:8]: 1 property per line makes the code more organized and readable

File: app/src/main/java/com/findmyrecycling/dto/User.kt [hunk 5]
```diff
@@ -1,4 +1,8 @@
 package com.findmyrecycling.dto
 
-class User (val uid: String = "", var displayName: String?) {
+class User (
+    val uid: String = "",
+    var displayName: String?
+    )
+{
```
Comment (JellyV) [inline app/src/main/java/com/findmyrecycling/dto/User.kt:7]: 1 property per line makes the code more organized and readable

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 6]
```diff
@@ -22,8 +22,8 @@ class ProductTests {
     @get:Rule
     var rule: TestRule = InstantTaskExecutorRule()
 
-    lateinit var productService : ProductService
-    var allProducts : List<Product>? = ArrayList<Product>()
+    private lateinit var productService : ProductService
+    private var allProducts : List<Product>? = ArrayList()
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:26]: Added private modifiers; Removed explicit type arguments

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 7]
```diff
@@ -130,7 +130,7 @@ class ProductTests {
 
     private fun thenResultsShouldContainCellPhoneRecycling() {
 
-        var allProducts : List<Product>? = ArrayList<Product>()
+        var allProducts : List<Product>? = ArrayList()
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:133]: Removed explicit type arguments

File: app/src/test/java/com/findmyrecycling/ProductUnitTests.kt [hunk 8]
```diff
@@ -68,7 +68,7 @@ class ProductUnitTests {
     }
 
     private fun thenResultsShouldContainNameOfRecyclingCenter() {
-        var allProducts : List<Product>? = ArrayList<Product>()
+        var allProducts : List<Product>? = ArrayList()
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductUnitTests.kt:71]: Removed explicit type arguments

File: app/src/test/java/com/findmyrecycling/ProductUnitTests.kt [hunk 9]
```diff
@@ -81,12 +81,12 @@ class ProductUnitTests {
...
-        var containsNameOfRecyclinCenter = false
+        var containsNameOfRecyclingCenter = false
         allProducts!!.forEach {
             if (it.product == (("Cell Phones"))) {
-                containsNameOfRecyclinCenter = true
+                containsNameOfRecyclingCenter = true
             }
         }
-        Assert.assertTrue(containsNameOfRecyclinCenter)
+        Assert.assertTrue(containsNameOfRecyclingCenter)
```
Comment (JellyV) [inline app/src/test/java/com/findmyrecycling/ProductUnitTests.kt:90]: Fixed typos

**Comment by discospiff** [comment]
From this review:
- Fix typos.  You don't need to merge the branch for this, though.  You can implement those manually.
Auto formatting and auto organize imports can be accomplished automatically, and consistently across all developers, with the IDE.  No need to do it manually in a branch, and then merge it.

**Pull request closed without merging**